### PR TITLE
suggestions related to DataFrames.jl

### PR DESCRIPTION
**Pull request by bkamins**
Hi @h-Klok,

I have made some suggestions to your codes. Please feel free to accept or reject them.
The changes are related to:
* not using deprecated functionality
* avoiding functionality that might change in the future
* some code clenup (how I would write things - but of course this is a matter of taste) => the main changes of this kind are in dataframeImputation.jl and scatterplot.jl in chapter 4

All the best with your book!

CC @nalimilan

### CODE REVIEW

File: 4_chapter/dataframeImputation.jl [hunk 1]
```diff
@@ -2,33 +2,20 @@ using Random, Statistics
 Random.seed!(0)
 include("dataframeCreation.jl")
 
-noTypePrice = (ismissing.(purchaseData[:Type])) .& 
-	(ismissing.(purchaseData[:Price]))
-purchaseData = purchaseData[.!noTypePrice,:]
-
-purchaseData[ismissing.(purchaseData[:Name]), :Name] = "notRecorded"
-purchaseData[ismissing.(purchaseData[:Time]), :Time] = "12:00 PM" 
+filter!(purchaseData) do row
```
Comment (nalimilan) [inline 4_chapter/dataframeImputation.jl:5]: Unfortunately, this is probably slower than the current code, right? We need to extend the pair API to `filter!` to make it efficient.

File: 4_chapter/summarizingData.jl [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+data = parse.(Int, readlines("grades.csv"))
 
 xbar = mean(data)
 svar = var(data)
 sdev = std(data)
 minval = minimum(data)
 maxval = maximum(data)
 med = median(data)
-per95 = percentile(data, 25) 
+per95 = percentile(data, 25)
```
Comment (nalimilan) [inline 4_chapter/summarizingData.jl:11]: Shouldn't this be `.95`?

File: 4_chapter/dataframeImputation.jl (hunk 1 above)
Comment (bkamins) [reply 4_chapter/dataframeImputation.jl:5]: Agreed. That is why I have said that all the changes are only suggestions. In this case I simply find `filter!` simpler to reason about and for normal statistics the performance difference is negligible.

**Comment by h-Klok** [comment]
Hi @bkamins and @nalimilan,
Thank you very much for your excellent suggestions, they are much appreciated and definitely worth incorporating!

**Comment by bkamins** [comment]
@h-Klok - you are welcome 😄. And all the best with your book.

**Pull request merged**